From ad76a6dd21910459502b2240b3fbb64bc735b817 Mon Sep 17 00:00:00 2001
From: shuall <shualloret@gmail.com>
Date: Fri, 23 Oct 2015 18:06:50 -0400
Subject: [PATCH] searchengines

---
 config.def.h |  5 +++++
 surf.c       | 23 +++++++++++++++++++++--
 2 files changed, 26 insertions(+), 2 deletions(-)

diff --git a/config.def.h b/config.def.h
index 5245129..483a5b3 100644
--- a/config.def.h
+++ b/config.def.h
@@ -139,3 +139,8 @@ static Button buttons[] = {
     { ClkAny,               0,          8,      navigate,       { .i = -1 } },
     { ClkAny,               0,          9,      navigate,       { .i = +1 } },
 };
+
+static SearchEngine searchengines[] = {
+  { "g",        "http://www.google.com/search?q=%s"   },
+};
+
diff --git a/surf.c b/surf.c
index 28bffef..6ac09bf 100644
--- a/surf.c
+++ b/surf.c
@@ -96,6 +96,11 @@ typedef struct {
 	regex_t re;
 } SiteStyle;
 
+typedef struct {
+	char *token;
+	char *uri;
+} SearchEngine;
+
 static Display *dpy;
 static Atom atoms[AtomLast];
 static Client *clients = NULL;
@@ -177,6 +182,7 @@ static void loaduri(Client *c, const Arg *arg);
 static void navigate(Client *c, const Arg *arg);
 static Client *newclient(void);
 static void newwindow(Client *c, const Arg *arg, gboolean noembed);
+static gchar *parseuri(const gchar *uri);
 static void pasteuri(GtkClipboard *clipboard, const char *text, gpointer d);
 static gboolean contextmenu(WebKitWebView *view, GtkWidget *menu,
                             WebKitHitTestResult *target, gboolean keyboard,
@@ -827,8 +833,7 @@ loaduri(Client *c, const Arg *arg)
 		u = g_strdup_printf("file://%s", rp);
 		free(rp);
 	} else {
-		u = g_strrstr(uri, "://") ? g_strdup(uri)
-		    : g_strdup_printf("http://%s", uri);
+		u = parseuri(uri);
 	}
 
 	setatom(c, AtomUri, uri);
@@ -1166,6 +1171,20 @@ menuactivate(GtkMenuItem *item, Client *c)
 	}
 }
 
+gchar *
+parseuri(const gchar *uri) {
+	guint i;
+
+	for (i = 0; i < LENGTH(searchengines); i++) {
+		if (searchengines[i].token == NULL || searchengines[i].uri == NULL || *(uri + strlen(searchengines[i].token)) != ' ')
+			continue;
+		if (g_str_has_prefix(uri, searchengines[i].token))
+			return g_strdup_printf(searchengines[i].uri, uri + strlen(searchengines[i].token) + 1);
+	}
+
+	return g_strrstr(uri, "://") ? g_strdup(uri) : g_strdup_printf("http://%s", uri);
+}
+
 void
 pasteuri(GtkClipboard *clipboard, const char *text, gpointer d)
 {
-- 
2.6.2

