diff --git a/slock.c b/slock.c
index face75e..33f4ec8 100644
--- a/slock.c.orig
+++ b/slock.c
@@ -189,6 +189,7 @@ unlockscreen(Display *dpy, Lock *lock) {
 static Lock *
 lockscreen(Display *dpy, int screen) {
 	char curs[] = {0, 0, 0, 0, 0, 0, 0, 0};
+	XClassHint *hint;
 	unsigned int len;
 	Lock *lock;
 	XColor color, dummy;
@@ -212,6 +213,10 @@ lockscreen(Display *dpy, int screen) {
 	lock->win = XCreateWindow(dpy, lock->root, 0, 0, DisplayWidth(dpy, lock->screen), DisplayHeight(dpy, lock->screen),
 			0, DefaultDepth(dpy, lock->screen), CopyFromParent,
 			DefaultVisual(dpy, lock->screen), CWOverrideRedirect | CWBackPixel, &wa);
+	hint = XAllocClassHint();
+	hint->res_name = "slock";
+	hint->res_class = "slock";
+	XSetClassHint(dpy,lock->win,hint);
 	XAllocNamedColor(dpy, DefaultColormap(dpy, lock->screen), COLOR2, &color, &dummy);
 	lock->colors[1] = color.pixel;
 	XAllocNamedColor(dpy, DefaultColormap(dpy, lock->screen), COLOR1, &color, &dummy);
