From c649bc365d2200873583592e814472fbe6eccecb Mon Sep 17 00:00:00 2001
From: shua <itis@isthisa.email>
Date: Wed, 15 Jun 2022 17:05:18 -0400
Subject: [PATCH] altkbd

---
 config.def.h | 10 +++++-----
 dwl.c        | 23 ++++++++++++++++++++++-
 2 files changed, 27 insertions(+), 6 deletions(-)

diff --git a/config.def.h b/config.def.h
index 43f35cd..b6719be 100644
--- a/config.def.h
+++ b/config.def.h
@@ -36,13 +36,13 @@ static const MonitorRule monrules[] = {
 };
 
 /* keyboard */
-static const struct xkb_rule_names xkb_rules = {
+static const struct xkb_rule_names xkb_rules[] = { {
 	/* can specify fields: rules, model, layout, variant, options */
+	/* multiple rules will be cycled through with nextkeymap */
 	/* example:
-	.options = "ctrl:nocaps",
+	{ .options = "ctrl:nocaps" },
 	*/
-	.options = NULL,
-};
+} };
 
 static const int repeat_rate = 25;
 static const int repeat_delay = 600;
@@ -116,7 +116,7 @@ static const Key keys[] = {
 	{ MODKEY,                    XKB_KEY_t,          setlayout,      {.v = &layouts[0]} },
 	{ MODKEY,                    XKB_KEY_f,          setlayout,      {.v = &layouts[1]} },
 	{ MODKEY,                    XKB_KEY_m,          setlayout,      {.v = &layouts[2]} },
-	{ MODKEY,                    XKB_KEY_space,      setlayout,      {0} },
+	{ MODKEY,                    XKB_KEY_space,      nextkeymap,     {.i = +1} },
 	{ MODKEY|WLR_MODIFIER_SHIFT, XKB_KEY_space,      togglefloating, {0} },
 	{ MODKEY,                    XKB_KEY_e,         togglefullscreen, {0} },
 	{ MODKEY,                    XKB_KEY_0,          view,           {.ui = ~0} },
diff --git a/dwl.c b/dwl.c
index 8a85c79..4526586 100644
--- a/dwl.c
+++ b/dwl.c
@@ -250,6 +250,7 @@ static void motionabsolute(struct wl_listener *listener, void *data);
 static void motionnotify(uint32_t time);
 static void motionrelative(struct wl_listener *listener, void *data);
 static void moveresize(const Arg *arg);
+static void nextkeymap(const Arg *arg);
 static void outputmgrapply(struct wl_listener *listener, void *data);
 static void outputmgrapplyortest(struct wlr_output_configuration_v1 *config, int test);
 static void outputmgrtest(struct wl_listener *listener, void *data);
@@ -325,6 +326,7 @@ static struct wl_list keyboards;
 static unsigned int cursor_mode;
 static Client *grabc;
 static int grabcx, grabcy; /* client-relative */
+static int xkb_rules_cur = 0;
 
 static struct wlr_output_layout *output_layout;
 static struct wlr_box sgeom;
@@ -811,7 +813,7 @@ createkeyboard(struct wlr_input_device *device)
 
 	/* Prepare an XKB keymap and assign it to the keyboard. */
 	context = xkb_context_new(XKB_CONTEXT_NO_FLAGS);
-	keymap = xkb_keymap_new_from_names(context, &xkb_rules,
+	keymap = xkb_keymap_new_from_names(context, xkb_rules,
 		XKB_KEYMAP_COMPILE_NO_FLAGS);
 
 	wlr_keyboard_set_keymap(device->keyboard, keymap);
@@ -1543,6 +1545,25 @@ moveresize(const Arg *arg)
 	}
 }
 
+void
+nextkeymap(const Arg *arg)
+{
+	Keyboard *kb;
+	xkb_rules_cur = (xkb_rules_cur + arg->i) % LENGTH(xkb_rules);
+	const struct xkb_rule_names *xkb_rule = &(xkb_rules[xkb_rules_cur]);
+
+	wl_list_for_each(kb, &keyboards, link) {
+		struct xkb_context *context = xkb_context_new(XKB_CONTEXT_NO_FLAGS);
+		struct xkb_keymap *keymap = xkb_map_new_from_names(context, xkb_rule,
+				XKB_KEYMAP_COMPILE_NO_FLAGS);
+		wlr_keyboard_set_keymap(kb->device->keyboard, keymap);
+		xkb_keymap_unref(keymap);
+		xkb_context_unref(context);
+	}
+	printf("kbd %s %s\n", xkb_rule->layout, xkb_rule->variant);
+	fflush(stdout);
+}
+
 void
 outputmgrapply(struct wl_listener *listener, void *data)
 {
-- 
2.36.1

