From 004fce6ed9cd048b2232906c28985d499ebe11cf Mon Sep 17 00:00:00 2001
From: shua <gpg@isthisa.email>
Date: Sun, 3 Nov 2024 22:00:27 +0100
Subject: [PATCH 1/4] make protocols

---
 config.mk | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/config.mk b/config.mk
index e2f1816..09f0f5a 100644
--- a/config.mk
+++ b/config.mk
@@ -24,13 +24,11 @@ WLR_LIBS = `$(PKG_CONFIG) --libs wlroots-0.19`
 #	-I$(PWD)/wlroots/0.19/include/wlroots-0.19
 #WLR_LIBS = -Wl,-rpath,$(PWD)/wlroots/0.19/lib64 -L$(PWD)/wlroots/0.19/lib64 -lwlroots-0.19
 
-XWAYLAND =
-XLIBS =
 # Uncomment to build XWayland support
-#XWAYLAND = -DXWAYLAND
-#XLIBS = xcb xcb-icccm
+XWAYLAND = -DXWAYLAND
+XLIBS = xcb xcb-icccm
 
 # dwl itself only uses C99 features, but wlroots' headers use anonymous unions (C11).
 # To avoid warnings about them, we do not use -std=c99 and instead of using the
 # gmake default 'CC=c99', we use cc.
-CC = cc
+CC = clang
-- 
2.47.0

