From f859a0cbaf34c35a2b145aa89f762691c99bb95a Mon Sep 17 00:00:00 2001
From: shua <gpg@isthisa.email>
Date: Sun, 28 Jul 2024 15:01:36 +0200
Subject: [PATCH 1/4] make protocols

---
 config.mk | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/config.mk b/config.mk
index c5e3b4c..41dfaa2 100644
--- a/config.mk
+++ b/config.mk
@@ -8,10 +8,8 @@ PREFIX = /usr/local
 MANDIR = $(PREFIX)/share/man
 DATADIR = $(PREFIX)/share
 
-XWAYLAND =
-XLIBS =
 # Uncomment to build XWayland support
-#XWAYLAND = -DXWAYLAND
-#XLIBS = xcb xcb-icccm
+XWAYLAND = -DXWAYLAND
+XLIBS = xcb xcb-icccm
 
-CC = gcc
+CC = clang
-- 
2.45.2

